### ISSUE TITLE
A master should be set to slave only after the MDS config has been updated

### ISSUE THREAD

**Issue by khenderick** [openvstorage/framework#971]
Assume following scenario for a given vDisk.

```
mds = [1, 2, 3]
```

if for some reason we want to swap 1 and 2 so 2 becomes the new master following scenario is followed:

```
first = [2, 3]
second = [2, 1, 3] 
1.setrole(slave)
3.setrole(slave)
update_mds_config(first)
update_mds_config(second)
```

This causes the current master `1` to become slave while it's still supposed to be master. So in an ideal flow, it should be sth like

```
first = [2, 3]
second = [2, 1, 3]
3.setrole(slave)
update_mds_config(first)
1.setrole(slave)
update_mds_config(second)
```

**Comment by khenderick** [comment]
Fixed by #979, packaged in openvstorage-2.7.3-rev.4059.6d404af

**Comment by JeffreyDevloo** [comment]
### Information

Conducting this flow manually is hard labor where surely a lot of numerous amount of details wouldn't be checked.
Therefore I will rely on the unit testing provided by Engineer to test this.
The unit tests will ensure that future changes will incorporate this flow of demote/promote.
Solely testing whether the unit tests do not fail on a fresh environment that has this fix:

### Steps
- Run mds unit test

### Output

```


###########################################################################################
# Processing test-module /opt/OpenvStorage/ovs/lib/tests/mdsservice_tests/test_mdsservice #
###########################################################################################

test_ensure_safety_of_2 (test_mdsservice.MDSServices) ... ok
test_ensure_safety_of_3 (test_mdsservice.MDSServices) ... ok
test_load_calculation (test_mdsservice.MDSServices) ... ok
test_mds_checkup (test_mdsservice.MDSServices) ... ok
test_role_assignments (test_mdsservice.MDSServices) ... ok
test_storagedriver_config_set (test_mdsservice.MDSServices) ... ok
test_syncreality (test_mdsservice.MDSServices) ... ok

----------------------------------------------------------------------
Ran 7 tests in 10.506s

OK



############
# OVERVIEW #
############

  - TestModule: /opt/OpenvStorage/ovs/lib/tests/mdsservice_tests/test_mdsservice  (7 tests)
    - DURATION: 10 seconds
    - SUCCESS: 7
```

### Test result

Test passed.
